src/device/index.js

```diff
@@ -392,6 +406,74 @@ realityEditor.device.postEventIntoIframe = function(event, frameKey, nodeKey) {
             }
         }
     }
+    
+    const entries = [];
+    realityEditor.forEachFrameInAllObjects((object, frame) => {
+        entries.push({
+            object,
+            frame
+        });
+    });
+    let currentIndex = realityEditor.device.raycastIndex;
+    realityEditor.device.raycastIndex += 1;
+    const frameRaycastPromises = entries.map(entry => {
+        return new Promise((resolve, reject) => {
+            const {object, frame} = entry;
+            const targetIframe = document.getElementById('iframe' + frame);
+            if (!targetIframe || !targetIframe.contentWindow) { // Avatars don't have iframes
+                resolve(null);
+                return;
+            }
+            targetIframe.contentWindow.postMessage(JSON.stringify({
+                raycastRequest: {
+                    coords: {
+                        x: newCoords.x,
+                        y: newCoords.y
+                    },
+                    index: currentIndex
+                }
+            }), '*');
+            let resolved = false;
+
+            const key = object + frame + currentIndex;
+            setTimeout(() => {
+                realityEditor.device.raycastCallbacks.delete(key);
+                if (!resolved) {
+                    resolved = true;
+                    reject();
+                }
+            }, 200); // Reject automatically within 200 ms
```

Comment:
https://github.com/ptcrealitylab/vuforia-spatial-toolbox-userinterface/assets/111802681/3759d415-6593-47de-a46f-07b8907dfc66

Looks like when I draw very fast, it raycasts on the table first and then flies off. When I draw slower, it forms these squiggly lines. Is it due to this 200ms interval where we decide to listen for tool raycasts?

Reply:
Close, that's a side-effect of how the drawing tool draws that I overlooked when adding the caching mechanism on the object.js side in the edge-server. Removing the caching doesn't seem to cause any performance issues, so I just removed that instead. This is just here to ensure a badly-behaved tool doesn't cause tons of touch events to be delayed.

Reply:
Got it. I pulled that commit and no longer get this artifact.